src/Clapi/Relay.hs

```diff
@@ -166,3 +170,18 @@ handleUnclaimedMessages vs msgs = if isValidClaim then handleOwnerMessages vs ms
             _ -> "Top level path already claimed"
         _ -> "Bundle claims multiple toplevel paths"
     errorAll = map (\m -> MsgError (msgPath' m) "Not within known toplevel path") msgs
+
+handleClientMessages :: Valuespace Validated -> [Message] -> [Message]
+handleClientMessages vs msgs = rmsgs ++ subMsgs
+  where
+    subMsgs = concatMap createMsgs subPaths
+    createMsgs p = [assignMsg p] ++ nodeMsgs p
+    assignMsg p = case getType vs p of
+        (Just tp) -> MsgAssignType p tp
+    nodeMsgs p = map (treeDeltaToMsg p) $ rightOrDie $ Tree.nodeDiff mempty $ nodeFor p
+    nodeFor p = fromJust $ Map.lookup p $ vsGetTree vs
+    subPaths = msgPath' <$> filter isSub msgs
+    isSub (MsgSubscribe _) = True
+    isSub _ = False
+    rightOrDie (Right x) = x
```

Comment:
`rightOrDie` is just `fromRight`

Reply:
But for some reason I can't get at fromRight, and in any case it isn't because fromRight has a default.